flink-core/src/main/java/org/apache/flink/util/CollectionUtil.java

```diff
@@ -75,6 +75,7 @@ public static boolean isNullOrEmpty(Map<?, ?> map) {
 		for (T element : elements) {
 			int bucket = index % numBuckets;
 			buckets.computeIfAbsent(bucket, key -> new ArrayList<>(initialCapacity)).add(element);
+			index++;
```

Comment:
@sjwiesman Can you take care of adding a unit test?

Reply:
I've opened another PR for this fix that includes a test. https://github.com/apache/flink/pull/14408